Python/pylifecycle.c

```diff
@@ -632,7 +633,9 @@ pycore_create_interpreter(_PyRuntimeState *runtime,
         return status;
     }
 
-    const PyInterpreterConfig config = _PyInterpreterConfig_LEGACY_INIT;
+    PyInterpreterConfig config = _PyInterpreterConfig_LEGACY_INIT;
+    // The main interpreter always has its own GIL.
+    config.own_gil = 1;
```

Comment:
Presumably we're in a lot of trouble if the main interpreter exits first...

Do we have mechanisms to ensure it outlives any subinterpreters that share its GIL? Or perhaps we should reference count the GIL instead of marking ownership?

Reply:
Yeah, the main interpreter must be the last one.  There are many things that break if that isn't the case.